roles/docker/tasks/main.yml

```diff
@@ -24,12 +24,21 @@
   action: "{{ ansible_pkg_mgr }} name=docker{{ '-' + docker_version if docker_version is defined and docker_version != '' else '' }} state=present"
   when: not openshift.common.is_atomic | bool and not docker_version_result | skipped and docker_version_result.stdout | default('0.0', True) | version_compare(docker_version, 'lt')
 
+# Enable docker and record if it was changed, if it was changed then we have no
+# need to reset-failed.
+- name: Enable the docker service
+  service:
+    name: docker
+    enabled: yes
+  register: docker_enabled
```

Comment:
Instead of enabling here, can we just test for failure and only reset the failed state if it is in a failed state?